fix: add `@ton-community/func-js` to peerDependencies to fix missing module error

ya7on opened the pull request:
Add `@ton-community/func-js` to `peerDependencies`, so that downstream TypeScript projects can compile correctly when importing from `@ton/sandbox`.

### Problem

Currently, the package uses `@ton-community/func-js` in a `.d.ts` file, but it's listed only under `devDependencies`. This causes consumers to see an error when building with TypeScript.

**Example error in consuming project:**

```
> tsc

node_modules/@ton/sandbox/dist/debugger/DebugInfoCache.d.ts:2:44 - error TS2307: Cannot find module '@ton-community/func-js' or its corresponding type declarations.

2 import { DebugInfo as FuncDebugInfo } from '@ton-community/func-js';
                                             ~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error in node_modules/@ton/sandbox/dist/debugger/DebugInfoCache.d.ts:2
```

**Example `package.json` in the consumer:**

```json
"devDependencies": {
  "@types/node": "^24.0.13",
  "prettier": "^3.6.2",
  "ts-node": "^10.9.2",
  "typescript": "^5.8.3"
},
"dependencies": {
  "@ton/core": "^0.61.0",
  "@ton/crypto": "^3.3.0",
  "@ton/sandbox": "^0.35.1",
  "@ton/test-utils": "^0.9.0",
  "@ton/ton": "^15.3.1"
}
```

### Solution

This PR add the dependency to the proper section (`peerDependencies`) so it's available to projects that rely on `@ton/sandbox`.

### Tests

There is no way to write a test for this scenario, as it affects downstream TypeScript consumers and not runtime behavior.
The fix was verified manually by linking the local package into a separate project and confirming successful compilation.

## Checklist

Please ensure the following items are completed before requesting review:

* [X] Updated `CHANGELOG.md` with relevant changes
* [X] Documented the contribution in `README.md`

No documentation update required, internal dependency fix only.

* [X] Added tests to demonstrate correct behavior (both positive and negative cases)

There is no way to write a test for this scenario, as it affects downstream TypeScript consumers and not runtime behavior.

* [X] All tests pass successfully
* [X] Code passes linting checks (`yarn lint`)

krigga commented:
I wonder if maybe making it a peer dependency instead is a better solution? What do you think?

ya7on commented:
I've updated `package.json` to move the dependency to the `peedDependency` section. To ensure the package still builds i also added it back to `devDependencies`.

Could you please confirm is this okay with you?
